source/lac/scalapack.cc

```diff
@@ -323,117 +326,344 @@ void ScaLAPACKMatrix<NumberType>::invert()
 
 
 template <typename NumberType>
-std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenvalues_symmetric()
+std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenpairs_symmetric_by_index(const std::pair<unsigned int,unsigned int> &index_limits,
+    const bool compute_eigenvectors)
+{
+  // check validity of index limits
+  Assert (index_limits.first != numbers::invalid_unsigned_int,ExcMessage("index limits have invalid value"));
+  Assert (index_limits.second != numbers::invalid_unsigned_int,ExcMessage("index limits have invalid value"));
+  Assert (index_limits.first < (unsigned int)n_rows,ExcMessage("index limits have to be smaller than matrix size"));
+  Assert (index_limits.second < (unsigned int)n_rows,ExcMessage("index limits have to be smaller than matrix size"));
+
+  std::pair<unsigned int,unsigned int> idx = std::make_pair(std::min(index_limits.first,index_limits.second),
+                                                            std::max(index_limits.first,index_limits.second));
+
+  // compute all eigenvalues/eigenvectors
+  if (idx.first==0 && idx.second==(unsigned int)n_rows-1)
+    return eigenpairs_symmetric(compute_eigenvectors);
+  else
+    return eigenpairs_symmetric(compute_eigenvectors,idx);
+}
+
+
+
+template <typename NumberType>
+std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenpairs_symmetric_by_value(const std::pair<NumberType,NumberType> &value_limits,
+    const bool compute_eigenvectors)
+{
+  Assert (!std::isnan(value_limits.first),ExcMessage("value limits have invalid value"));
+  Assert (!std::isnan(value_limits.second),ExcMessage("value limits have invalid value"));
+
```

Comment:
maybe add a check that `values.first < values.second`?  (equality does not really make sense here anyway)

Reply:
Same as for one of the previous comments: `eigenpairs_symmetric` will use the correct ordering automatically